ogr/ogrsf_frmts/miramon/ogrmiramon.h

```diff
@@ -0,0 +1,173 @@
+/******************************************************************************
+ * $Id$
+ *
+ * Project:  OpenGIS Simple Features Reference Implementation
+ * Purpose:  C API to create a MiraMon layer
+ * Author:   Abel Pau
+ ******************************************************************************
+ * Copyright (c) 2024, Xavier Pons
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE. 
+ ****************************************************************************/
+
+#ifndef OGRMIRAMON_H_INCLUDED
+#define OGRMIRAMON_H_INCLUDED
+
+#include "ogrsf_frmts.h"
+#include "ogr_api.h"
+#include "cpl_string.h"
+#include "mm_wrlayr.h"
+
+/************************************************************************/
+/*                             OGRMiraMonLayer                          */
+/************************************************************************/
+
+class OGRMiraMonLayer final
+    : public OGRLayer,
+      public OGRGetNextFeatureThroughRaw<OGRMiraMonLayer>
+{
+    OGRSpatialReference *m_poSRS = nullptr;
+    OGRFeatureDefn *poFeatureDefn;
```

Comment:
inline default initialization of all those member variables would be better.
and using the m_ prefixing for all of them would also be better for consistency

Reply:
inline initialization will only work for primitive types though (pointers, integers, doubles, etc). For the C structures like struct MiraMonVectLayerInfo hMiraMonLayerPNT/ARC/POL/etc, memset() to 0 in the constructor is the only practical way

Reply:
- [x] inline default initialization of all those member variables would be better.
- [x] m_ prefixing for all of them would also be better for consistency
But the hMiraMonLayerPNT/ARC/POL/etc I'd prefer not to change now. They are at too many places.

- [x] inline initialization will only work for primitive types though (pointers, integers, doubles, etc). For the C structures like struct MiraMonVectLayerInfo hMiraMonLayerPNT/ARC/POL/etc, memset() to 0 in the constructor is the only practical way

In fact MMInitLayer() initializes everything. So no worries